### ISSUE TITLE
Tell users that if they claim from the farm instead of the givstream or givbacks page, they save on gas!

### ISSUE THREAD

**Issue by laurenluz** [Giveth/GIVeconomy#399]

**Comment by laurenluz** [comment]
maybe we do a quick check to see if the user has rewards available to harvest in the GIVfarm, and then, if so, there is a popup when they click harvest that says...

" You have GIVfarm rewards ready to claim! If you harvest rewards in the GIVfarm, we will send you your GIVstream and/or GIVbacks rewards with it in a single transaction with no extra gas."

Button: "Go to GIVfarm" links to: https://giv.giveth.io/givfarm
smaller text: "Harvest rewards here anyway" which opens the harvest modal

**Comment by laurenluz** [comment]
@markoprljic this is just something I noticed... what do you think? worth adding a popup/message? or not?

**Comment by markoprljic** [comment]
I didn't know about this scenario. Why is it that they save in gas when harvesting from the farm? Aren't we using the same harvest modal across the GIVeconomy, and isn't the user harvesting all rewards regardless from where they trigger the Harvest function? 

Curious to know why we can't harvest from the farm anyway regardless where user triggers the function and thus avoid the need to inform them and redirect them to a different page (if we can pull this on the backend without their awareness)?

**Comment by laurenluz** [comment]
When you harvest from the farm, you only harvest from one farm at a time, not all. Those rewards aren't *already* in the token distro. But when you harvest them, you also get everything sent to you from the token distro.

GIVbacks rewards, GIVdrop, GIVstream... they are all together in the token distro so harvesting is done in on tx.

It doesn't make sense to harvest all farms together both because it is challenging to implement and because on mainnet, the cumulative gas could become very expensive. There was an issue for this, we investigated w the devs, we decided it is not worth it.
